bundles/admin/admin-layereditor/view/AdminLayerForm/AdminLayerFormHandler.js

```diff
@@ -999,11 +999,36 @@ class UIHandler extends StateHandler {
 
         this.updateState({ layer });
     }
+
+    saveStyleToLayer (style, styleName, originalName) {
+        const layer = this.getState().layer;
+        if (originalName !== '') {
+            delete layer.options.styles[originalName];
+        }
+
+        layer.options.styles = {
+            ...layer.options.styles,
+            [styleName]: style
```

Comment:
If the user adds a style called "style 1", then adds a style "style 2", then renames "style 2" -> "style 1" the other style is overwritten and the layer will only have one style.

I suggest we don't use style names from user input. We need to generate an id for style at runtime and save the user input inside the style object as "title" or "label" or something like that. It would also fix problematic names like names with quotes or spaces that need escaping and tend to cause problems at some point.